buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTest.kt

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.gradlebuild.test.integrationtests
+
+import org.gradle.api.Named
+import org.gradle.api.Project
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.ProjectLayout
+import org.gradle.api.tasks.*
+import org.gradle.api.tasks.testing.Test
+import org.gradle.build.GradleDistribution
+import org.gradle.internal.os.OperatingSystem
+import org.gradle.process.CommandLineArgumentProvider
+import java.util.concurrent.Callable
+
+/**
+ * Base class for all tests that check the end-to-end behavior of a Gradle distribution.
+ */
+open class DistributionTest : Test() {
+
+    @get: Input
+    val operatingSystem: String by lazy {
+        val current = OperatingSystem.current()
+        // the version currently differs between our dev infrastructure, so we only track the name and the architecture
+        current.getName() + " " + System.getProperty("os.arch")
+    }
+
+    @Internal
+    val gradleInstallationForTest: GradleInstallationForTestEnvironmentProvider = GradleInstallationForTestEnvironmentProvider(project)
+
+    @Internal
+    val binaryDistributions: BinaryDistributions = BinaryDistributions(project.layout)
+
+    @Internal
+    val libsRepository: LibsRepositoryEnvironmentProvider = LibsRepositoryEnvironmentProvider(project.layout)
```

Comment:
:x: Should be
```kotlin
@Internal
val libsRepository =
    LibsRepositoryEnvironmentProvider(project.layout)
```

The type can be omitted.
Same thing for the two other internal properties above.

Reply:
So the difference is removing the type and having the actual value in a new line?

Reply:
Done.